target/linux/realtek/files-5.10/drivers/net/dsa/rtl83xx/rtl839x.c

```diff
@@ -1817,6 +1817,84 @@ void rtl839x_set_receive_management_action(int port, rma_ctrl_t type, action_typ
 	}
 }
 
+static void rtl839x_led_init(struct rtl838x_switch_priv *priv)
+{
+	int i, pos;
+	u32 v, set;
+	u64 pm_copper = 0, pm_fiber = 0;
+	u32 setlen;
+	const __be32 *led_set;
+	char set_name[9];
+	struct device_node *node;
+	const char *mode;
+
+	node = of_find_compatible_node(NULL, NULL, "realtek,rtl8390-leds");
+	if (!node) {
+		pr_info("%s No compatible LED node found\n", __func__);
+		return;
+	}
+
+	for (i= 0; i < priv->cpu_port; i++) {
+		pos = (i << 1) % 32;
+		sw_w32_mask(0x3 << pos, 0, RTL839X_LED_FIB_SET_SEL_CTRL(i));
+		sw_w32_mask(0x3 << pos, 0, RTL839X_LED_COPR_SET_SEL_CTRL(i));
+
+		if (!priv->ports[i].phy)
+			continue;
+
+		if (priv->ports[i].phy_is_integrated)
+		pm_fiber |= BIT_ULL(i);
+			else
+		pm_copper |= BIT_ULL(i);
+
+		set = priv->ports[i].led_set;
+		sw_w32_mask(0, set << pos, RTL839X_LED_COPR_SET_SEL_CTRL(i));
+		sw_w32_mask(0, set << pos, RTL839X_LED_FIB_SET_SEL_CTRL(i));
+	}
+
+	sw_w32(0, RTL839X_LED_SET0_0_CTRL);
+	sw_w32(0, RTL839X_LED_SET0_0_CTRL - 4);
```

Comment:
`RTL839X_LED_SET_0_1_CTRL` and `RTL839X_LED_SET_2_3_CTRL`

Reply:
I need RTL839X_LED_SET0_0_CTRL also in the loop below, which uses the same logic to access the LED-set registers. The idea was to stay consistent.

Reply:
Then maybe change to a macro to calculate the correct offset. You do this anyway for other registers.